paddle/fluid/platform/device_context.cc

```diff
@@ -33,9 +33,15 @@ DeviceContextPool::DeviceContextPool(
   PADDLE_ENFORCE_GT(places.size(), 0);
   for (size_t i = 0; i < places.size(); i++) {
     if (platform::is_cpu_place(places[i])) {
+#ifdef PADDLE_WITH_MKLDNN
+      device_contexts_.emplace(places[i],
+                               new platform::MKLDNNDeviceContext(
+                                   boost::get<platform::CPUPlace>(places[i])));
+#else
       device_contexts_.emplace(places[i],
                                new platform::CPUDeviceContext(
                                    boost::get<platform::CPUPlace>(places[i])));
+#endif
```

Comment:
Honestly, I have a little question here. 
When `PADDLE_WITH_MKLDNN` is enabled, we will not have `platform::CPUDeviceContext` anyway .
Is that fine with you @jacquesqiao @QiJune ?

Reply:
@tensor-tang It seems that if `PADDLE_WITH_GPU` is enabled, there will be both `CPUDeviceContext` and `CUDADeviceContext`.
So, I think MKLDNNDeviceContext and CPUDeviceContext should be coexist.

Could you provide an example that has a two FC in the network, which one FC is CPU, and the other FC is MKLDNN? Just mnist demo will be fine.

We can see if "MKLDNN" is compatible with "CPU".

Reply:
Firstly, `MKLDNNDeviceContext`  is inherited from `CPUDeviceContext` now
 https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/platform/device_context.h#L113. So functionally, it can pass the CI on this version.

But from my perspective, I thought `CPUDeviceContext` should be always available no matter which third-party is added, not only MKLDNN here. Since we can not guarantee all the ops paddle supported are also supported by the third-party library.  If this third-party context is not inherited from cpu context, it would be a problem.

So I just want to hear you voice. I am not sure is that proper.

Reply:
Yes, I think so. 
Since `MKLDNNDeviceContext` is inherited from `CPUDeviceContext`, there will be no problem.
I have not think out a better choice, we can move on first.

Reply:
OK, Thx